Mathlib/Data/Nat/BitIndices.lean

```diff
@@ -0,0 +1,96 @@
+/-
+Copyright (c) 2024 Peter Nelson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Peter Nelson
+-/
+import Mathlib.Combinatorics.Colex
+
+/-!
+# Bit Indices
+
+Given `n : ℕ`, we define `Nat.bitIndices n`, which is the `Finset` of indices of `1`s in the
+binary expansion of `n`. We show that this function is the inverse of `fun s ↦ ∑ i in s, 2^i`,
+and therefore that it gives an order isomorphism from `ℕ` to `Colex ℕ`.
+
+The latter proof uses the machinery in `Combinatorics.Colex`, which is why this doesn't appear
+in an earlier file. The implementation of `Nat.bitIndices` is a direct recursive definition,
+which is definitionally cleaner than the other option of using `Nat.bits` and the `List` API.
+
+TODO : Relate the material in this file to `Nat.digits` and `Nat.bits`.
+-/
+
+open Finset BigOperators
+
+namespace Nat
+
+variable {a n : ℕ} {s : Finset ℕ}
+
+/-- The function which maps the natural number `∑ i in s, 2^i` to the Finset `s`. -/
+def bitIndices (n : ℕ) : Finset ℕ := by
```

Comment:
Should this be a (sorted) list instead of a the finest? Nothing about the name suggests that it should be a finset.

Reply:
In the context of colex, it should be a finset. But that doesn't mean we can't construct it as a list originally.